### PR TITLE
feat: cli: get tipset keys by height

### PR DESCRIPTION
**Pull request by jennijuju**
./lotus chain gettipset --height 9316
: [bafy2bzacecwefa7e7ryue6awxppqetr566katdb7zdu75t65dlr3nutf4zqf4]

### CODE REVIEW

File: cli/chain.go [hunk 1]
```diff
@@ -1504,3 +1505,38 @@ var ChainPruneCmd = &cli.Command{
 		return api.ChainPrune(ctx, opts)
 	},
 }
+
+var ChainGetTipSetByHeightCmd = &cli.Command{
+	Name:    "gettipset",
```
Comment (arajasek) [inline cli/chain.go:1510]:
```suggestion
	Name:    "get-tipset",
```

File: cli/chain.go [hunk 2]
```diff
@@ -1504,3 +1505,38 @@ var ChainPruneCmd = &cli.Command{
...
+	Aliases: []string{"love"},
```
Comment (arajasek) [inline cli/chain.go:1511]: ?

Comment (arajasek) [inline cli/chain.go:1511]:
```suggestion
	Aliases:   []string{"getblock"},
```

File: cli/chain.go [hunk 3]
```diff
@@ -1504,3 +1505,38 @@ var ChainPruneCmd = &cli.Command{
...
+	Usage:   "View a segment of the chain",
+	Flags: []cli.Flag{
+		&cli.Uint64Flag{Name: "height", DefaultText: "current head"},
```
Comment (arajasek) [inline cli/chain.go:1514]: I would suggest making this a mandatory parameter. We already have `lotus chain head` to print the head.

File: cli/chain.go [hunk 4]
```diff
@@ -1504,3 +1505,38 @@ var ChainPruneCmd = &cli.Command{
...
+		if cctx.IsSet("height") {
+			head, err = api.ChainGetTipSetByHeight(ctx, abi.ChainEpoch(cctx.Uint64("height")), types.EmptyTSK)
+		} else {
+			head, err = api.ChainHead(ctx)
+		}
+		if err != nil {
+			return err
+		}
+
+		head, err = api.ChainGetTipSet(ctx, head.Parents())
```
Comment (arajasek) [inline cli/chain.go:1535]: Hmm, I'm not sure I agree with printing info about the parents here. That might match some of our Ethereum-modelling behaviour, but it is just wrong to say the tipset at this height is the parent here.

File: cli/chain.go [hunk 5]
```diff
@@ -1504,3 +1505,38 @@ var ChainPruneCmd = &cli.Command{
...
+		if err != nil {
+			return err
+		}
+		fmt.Printf(": %s\n", head.Key().Cids())
```
Comment (raulk) [inline cli/chain.go:1539]: Can we print out the whole tipset in addition to the key?

File: cli/chain.go [hunk 6]
```diff
@@ -1504,3 +1505,38 @@ var ChainPruneCmd = &cli.Command{
 		return api.ChainPrune(ctx, opts)
 	},
 }
+
+var ChainGetTipSetByHeightCmd = &cli.Command{
+	Name:    "gettipset",
+	Aliases: []string{"love"},
+	Usage:   "View a segment of the chain",
```
Comment (arajasek) [inline cli/chain.go:1512]:
```suggestion
	Usage:   "Get a tipset at the specified height, and print the CIDs of the block headers"
```

**Pull request closed without merging**